pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/reader/provider/IndexReaderProvider.java

```diff
@@ -0,0 +1,25 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.segment.spi.index.reader.provider;
+
+public interface IndexReaderProvider
```

Comment:
Do we need to have separate provider interfaces for each index type since the pluggable unit is `IndexReaderProvider`? I feel it might be more readable if we just put all methods into this single interface?

Reply:
I prefer it this way because it allows implementations in pinot-core/pinot-segment-local to only depend on functionality they need, which improves testability/modularity. The registration mechanism conflating these index types together is an implementation detail for the convenience of the user.